### ISSUE TITLE
cargo::tests::get_crate_file_from_overrides test is broken on Windows

### ISSUE THREAD

**Issue by Arnavion** [phildawes/racer#631]
```
---- cargo::tests::get_crate_file_from_overrides stdout ----
        thread 'cargo::tests::get_crate_file_from_overrides' panicked at 'assertion failed: `(left == right)` (left: `"\\\\?\\C:\\Stuff\\Sources\\racer\\fixtures\\arst\\src\\lib.rs"`, right: `"C:\\Stuff\\Sources\\racer\\fixtures\\arst\\src\\lib.rs"`)', src/racer\cargo.rs:678
stack backtrace:
   0:        0x1402ab86e - std::panicking::rust_panic_with_hook::h4cbd7ca63ce1aee9
   1:        0x1402aa1f1 - std::rt::lang_start::h53bf99b0829cc03c
   2:        0x1402aad4d - std::panicking::rust_panic_with_hook::h4cbd7ca63ce1aee9
   3:        0x1402aab96 - std::panicking::begin_panic_fmt::hd0daa02942245d81
   4:        0x1402aaaf4 - std::panicking::begin_panic_fmt::hd0daa02942245d81
   5:        0x13ff3b522 - racer::cargo::tests::get_crate_file_from_overrides
                        at C:\Stuff\Sources\racer\<std macros>:8
   6:        0x1401a427a - term::terminfo::parser::compiled::parse::{{closure}}
   7:        0x14019c456 - term::terminfo::parser::compiled::parse::{{closure}}
   8:        0x1402ada31 - _rust_maybe_catch_panic
   9:        0x1401a3a04 - term::terminfo::parser::compiled::parse::{{closure}}
  10:        0x1402a886e - std::sys::thread::Thread::new::h47f2f5af38ea3f4d
  11:         0x777259cc - BaseThreadInitThunk


failures:
    cargo::tests::get_crate_file_from_overrides
```

`PathBuf::canonicalize` is implemented using [GetFinalPathNameByHandle](https://msdn.microsoft.com/en-us/library/windows/desktop/aa364962), and as that pages says `The string that is returned by this function uses the \\?\ syntax.`

**Comment by jwilm** [comment]
I just merged #623; do you have that change?

**Comment by Arnavion** [comment]
Ah, sorry about the noise.
